diff --git a/lib/zip_random_unix.c b/lib/zip_random_unix.c
index f84a5482..55379566 100644
--- a/lib/zip_random_unix.c
+++ b/lib/zip_random_unix.c
@@ -94,10 +94,10 @@ zip_random_uint32(void) {
     }
 
     if (!seeded) {
-        srandom((unsigned int)time(NULL));
+        srand((unsigned int)time(NULL));
     }
 
-    return (zip_uint32_t)random();
+    return (zip_uint32_t)rand();
 }
 #endif
 
