diff -ruN a/configure.ac b/configure.ac
--- a/configure.ac	2016-05-23 12:01:08.000000000 +0200
+++ b/configure.ac	2017-01-24 21:48:55.752848317 +0100
@@ -216,6 +216,11 @@
 fi
 AM_CONDITIONAL([REBUILD_DOCS], [test "$enable_rebuild_docs" = "yes" -o "$USER" = "veillard"])
 
+# Patch from clutter-android
+AC_ARG_ENABLE(tests,
+[  --enable-tests[[=yes/no]]  build tests [[default=yes]]])
+AM_CONDITIONAL([BUILD_TESTS], [test "$enable_tests" = "yes"])
+
 dnl
 dnl hard dependancies on options
 dnl
diff -ruN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2016-05-06 18:28:03.000000000 +0200
+++ b/Makefile.am	2017-01-24 21:49:23.702847706 +0100
@@ -10,10 +10,15 @@
 
 AM_CFLAGS = $(THREAD_CFLAGS) $(Z_CFLAGS) $(LZMA_CFLAGS)
 
+## Patch from clutter-android
+if BUILD_TESTS
 noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
                 testThreads testC14N testAutomata testRegexp \
                 testReader testapi testModule runtest runsuite testchar \
 		testdict runxmlconf testrecurse testlimits
+else
+noinst_PROGRAMS=
+endif
 
 bin_PROGRAMS = xmllint xmlcatalog
 
diff -ruN a/xmlIO.c b/xmlIO.c
--- a/xmlIO.c	2016-10-12 18:25:04.000000000 +0200
+++ b/xmlIO.c	2017-01-24 21:09:25.482900125 +0100
@@ -47,6 +47,10 @@
 #include <winnls.h> /* for CP_UTF8 */
 #endif
 
+#if defined(__vita__)
+#define getcwd(buf, size) NULL
+#endif
+
 /* Figure a portable way to know if a file is a directory. */
 #ifndef HAVE_STAT
 #  ifdef HAVE__STAT
