diff --git a/icu/source/common/putilimp.h b/icu/source/common/putilimp.h
index e5473225b0..c148a32ae2 100644
--- a/icu/source/common/putilimp.h
+++ b/icu/source/common/putilimp.h
@@ -112,6 +112,7 @@ typedef size_t uintptr_t;
 #endif
 #elif U_PLATFORM == U_PF_OS400
    /* not defined */
+#elif defined(__vita__)
 #else
 #   define U_TZSET tzset
 #endif
@@ -150,6 +151,7 @@ typedef size_t uintptr_t;
 #endif
 #elif U_PLATFORM == U_PF_OS400
    /* not defined */
+#elif defined(__vita__)
 #else
 #   define U_TZNAME tzname
 #endif
diff --git a/icu/source/i18n/digitlst.cpp b/icu/source/i18n/digitlst.cpp
index 4d7c3f4acb..2292c01283 100644
--- a/icu/source/i18n/digitlst.cpp
+++ b/icu/source/i18n/digitlst.cpp
@@ -64,7 +64,7 @@
 # if U_PLATFORM_USES_ONLY_WIN32_API || U_PLATFORM == U_PF_CYGWIN
 #   include <locale.h>
 # else
-#   include <xlocale.h>
+#   include <locale.h>
 # endif
 #endif
 
diff --git a/icu/source/tools/toolutil/toolutil.cpp b/icu/source/tools/toolutil/toolutil.cpp
index fb7263dcfe..ebcf73a882 100644
--- a/icu/source/tools/toolutil/toolutil.cpp
+++ b/icu/source/tools/toolutil/toolutil.cpp
@@ -67,6 +67,10 @@
 #include "toolutil.h"
 #include "unicode/ucal.h"
 
+#if defined(__vita__)
+#include <psp2/io/stat.h>
+#endif
+
 U_NAMESPACE_BEGIN
 
 IcuToolErrorCode::~IcuToolErrorCode() {
@@ -188,6 +192,8 @@ uprv_mkdir(const char *pathname, UErrorCode *status) {
     int retVal = 0;
 #if U_PLATFORM_USES_ONLY_WIN32_API
     retVal = _mkdir(pathname);
+#elif defined(__vita__)
+    retVal = sceIoMkdir(pathname, S_IRWXU | (S_IROTH | S_IXOTH) | (S_IROTH | S_IXOTH));
 #else
     retVal = mkdir(pathname, S_IRWXU | (S_IROTH | S_IXOTH) | (S_IROTH | S_IXOTH));
 #endif

diff --git a/icu/source/common/unicode/putil.h b/icu/source/common/unicode/putil.h
index 406551a8d8..523dc50d0d 100644
--- a/icu/source/common/unicode/putil.h
+++ b/icu/source/common/unicode/putil.h
@@ -126,6 +126,13 @@ U_INTERNAL void U_EXPORT2 u_setTimeZoneFilesDirectory(const char *path, UErrorCo
 #   define U_FILE_SEP_STRING "\\"
 #   define U_FILE_ALT_SEP_STRING "/"
 #   define U_PATH_SEP_STRING ";"
+#elif defined(__vita__)
+#   define U_FILE_SEP_CHAR '/'
+#   define U_FILE_ALT_SEP_CHAR '/'
+#   define U_PATH_SEP_CHAR ';'
+#   define U_FILE_SEP_STRING "/"
+#   define U_FILE_ALT_SEP_STRING "/"
+#   define U_PATH_SEP_STRING ";"
 #else
 #   define U_FILE_SEP_CHAR '/'
 #   define U_FILE_ALT_SEP_CHAR '/'
