From 1e49af0b2dd7b94f42c2d17eed4d9be832e3cd93 Mon Sep 17 00:00:00 2001
From: devnoname120 <devnoname120+gh@gmail.com>
Date: Fri, 26 May 2017 16:57:24 +0200
Subject: [PATCH 1/2] Vita changes

---
 Makefile | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9754ddf..2b2b48d 100644
--- a/Makefile
+++ b/Makefile
@@ -15,9 +15,9 @@
 SHELL=/bin/sh
 
 # To assist in cross-compiling
-CC=gcc
-AR=ar
-RANLIB=ranlib
+CC=arm-vita-eabi-gcc
+AR=arm-vita-eabi-ar
+RANLIB=arm-vita-eabi-ranlib
 LDFLAGS=
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
@@ -69,6 +69,10 @@ test: bzip2
 	cmp sample3.tst sample3.ref
 	@cat words3
 
+installlib: libbz2.a
+	cp -f libbz2.a $(PREFIX)/lib
+	cp -f bzlib.h $(PREFIX)/include
+
 install: bzip2 bzip2recover
 	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
 	if ( test ! -d $(PREFIX)/lib ) ; then mkdir -p $(PREFIX)/lib ; fi
-- 
2.13.0


From e322ca9f25de615e75b3668ee8e14bacf14ee6d4 Mon Sep 17 00:00:00 2001
From: devnoname120 <devnoname120+gh@gmail.com>
Date: Fri, 2 Jun 2017 19:41:12 +0200
Subject: [PATCH 2/2] Fix installlib recipe

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 2b2b48d..6c7b4f2 100644
--- a/Makefile
+++ b/Makefile
@@ -70,8 +70,8 @@ test: bzip2
 	@cat words3
 
 installlib: libbz2.a
-	cp -f libbz2.a $(PREFIX)/lib
-	cp -f bzlib.h $(PREFIX)/include
+	cp -f libbz2.a $(PREFIX)/lib/
+	cp -f bzlib.h $(PREFIX)/include/
 
 install: bzip2 bzip2recover
 	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
-- 
2.13.0

