From c3700e6504247a246938513e9b5d85b67e171f35 Mon Sep 17 00:00:00 2001
From: devnoname120 <devnoname120+gh@gmail.com>
Date: Fri, 26 May 2017 16:57:24 +0200
Subject: [PATCH] Vita changes

---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9754ddf..580bc0d 100644
--- a/Makefile
+++ b/Makefile
@@ -15,9 +15,9 @@
 SHELL=/bin/sh
 
 # To assist in cross-compiling
-CC=gcc
-AR=ar
-RANLIB=ranlib
+CC=arm-vita-eabi-gcc
+AR=arm-vita-eabi-ar
+RANLIB=arm-vita-eabi-ranlib
 LDFLAGS=
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
-- 
2.13.0

