diff --git a/Makefile.vita b/Makefile.vita
new file mode 100644
index 0000000..9f9eb98
--- /dev/null
+++ b/Makefile.vita
@@ -0,0 +1,26 @@
+TARGET    = libunrar.a
+OBJS      = archive.o arcread.o blake2s.o cmddata.o consio.o crc.o crypt.o \
+			dll.o encname.o errhnd.o extinfo.o extract.o filcreat.o file.o \
+			filefn.o filestr.o find.o getbits.o global.o hash.o headers.o \
+			match.o options.o pathfn.o qopen.o rar.o rarpch.o rarvm.o \
+			rawread.o rdwrfn.o rijndael.o rs.o rs16.o scantree.o secpassword.o \
+			sha1.o sha256.o smallfn.o strfn.o strlist.o system.o threadpool.o \
+			timefn.o ui.o unicode.o unpack.o volume.o
+
+PREFIX   = arm-vita-eabi
+CC       = $(PREFIX)-gcc
+CXX      = $(PREFIX)-g++
+AR       = $(PREFIX)-ar
+CFLAGS   = -Wl,-q -Wall -O3 -Wno-unused-variable -Wno-unused-but-set-variable -ffat-lto-objects -flto -DRARDLL -DUNRAR -DSILENT
+CXXFLAGS = $(CFLAGS) -std=gnu++11 -fno-rtti
+
+.PHONY: all clean
+
+all: $(TARGET)
+
+$(TARGET): $(OBJS)
+	$(AR) rcu $@ $^
+
+clean:
+	@rm -rf $(TARGET) $(OBJS)
+	@echo "Cleaned up!"
diff --git a/arcread.cpp b/arcread.cpp
index 7cfa30b..a90884d 100644
--- a/arcread.cpp
+++ b/arcread.cpp
@@ -1366,6 +1366,7 @@ void Archive::ConvertAttributes()
   // We restore native Unix attributes as is, because it can be backup.
   static mode_t mask = (mode_t) -1;
 
+#ifndef __vita__
   if (mask == (mode_t) -1)
   {
     // umask call returns the current umask value. Argument (022) is not
@@ -1375,6 +1376,9 @@ void Archive::ConvertAttributes()
     // Restore the original umask value, which was changed to 022 above.
     umask(mask);
   }
+#else
+  mask = 022;
+#endif
 
   switch(FileHead.HSType)
   {
diff --git a/extinfo.cpp b/extinfo.cpp
index 614ca89..1c41acd 100644
--- a/extinfo.cpp
+++ b/extinfo.cpp
@@ -9,7 +9,9 @@
 #endif
 
 #ifdef _UNIX
+#ifndef __vita__
 #include "uowners.cpp"
+#endif
 #ifdef SAVE_LINKS
 #include "ulinks.cpp"
 #endif
@@ -43,10 +45,12 @@ void SetExtraInfo20(CommandData *Cmd,Archive &Arc,const std::wstring &Name)
 void SetExtraInfo(CommandData *Cmd,Archive &Arc,const std::wstring &Name)
 {
 #ifdef _UNIX
+#ifndef __vita__
   if (!Cmd->Test && Cmd->ProcessOwners && Arc.Format==RARFMT15 &&
       Arc.SubHead.CmpName(SUBHEAD_TYPE_UOWNER))
     ExtractUnixOwner30(Arc,Name.c_str());
 #endif
+#endif
 #ifdef _WIN_ALL
   if (!Cmd->Test && Cmd->ProcessOwners && Arc.SubHead.CmpName(SUBHEAD_TYPE_ACL))
     ExtractACL(Arc,Name);
@@ -60,9 +64,11 @@ void SetExtraInfo(CommandData *Cmd,Archive &Arc,const std::wstring &Name)
 void SetFileHeaderExtra(CommandData *Cmd,Archive &Arc,const std::wstring &Name)
 {
 #ifdef _UNIX
+#ifndef __vita__
    if (Cmd->ProcessOwners && Arc.Format==RARFMT50 && Arc.FileHead.UnixOwnerSet)
      SetUnixOwner(Arc,Name);
 #endif
+#endif
 }
 
 
diff --git a/file.cpp b/file.cpp
index c6fab17..ee14746 100644
--- a/file.cpp
+++ b/file.cpp
@@ -121,6 +121,7 @@ bool File::Open(const std::wstring &Name,uint Mode)
   WideToChar(Name,NameA);
 
   int handle=open(NameA.c_str(),flags);
+#ifndef __vita__
 #ifdef LOCK_EX
 
 #ifdef _OSF_SOURCE
@@ -132,6 +133,7 @@ bool File::Open(const std::wstring &Name,uint Mode)
     return false;
   }
 
+#endif
 #endif
   if (handle==-1)
     hNewFile=FILE_BAD_HANDLE;
@@ -714,6 +716,7 @@ void File::SetCloseFileTime(RarTime *ftm,RarTime *fta)
 
 void File::SetCloseFileTimeByName(const std::wstring &Name,RarTime *ftm,RarTime *fta)
 {
+#ifndef __vita__
 #ifdef _UNIX
   bool setm=ftm!=NULL && ftm->IsSet();
   bool seta=fta!=NULL && fta->IsSet();
@@ -743,6 +746,7 @@ void File::SetCloseFileTimeByName(const std::wstring &Name,RarTime *ftm,RarTime
 #endif
   }
 #endif
+#endif
 }
 
 
diff --git a/filefn.cpp b/filefn.cpp
index 1f34d26..fbe9b52 100644
--- a/filefn.cpp
+++ b/filefn.cpp
@@ -157,6 +157,7 @@ int64 GetFreeDisk(const std::wstring &Name)
       uiUserFree.u.HighPart<=uiTotalFree.u.HighPart)
     return INT32TO64(uiUserFree.u.HighPart,uiUserFree.u.LowPart);
   return 0;
+#elif defined(__vita__)
 #elif defined(_UNIX)
   std::wstring Root;
   GetPathWithSep(Name,Root);
diff --git a/os.hpp b/os.hpp
index f94b5b3..6780151 100644
--- a/os.hpp
+++ b/os.hpp
@@ -131,8 +131,10 @@
   #include <sys/sysctl.h>
 #endif
 #ifndef SFX_MODULE
+#ifndef __vita__
     #include <sys/statvfs.h>
 #endif
+#endif
 #include <pwd.h>
 #include <grp.h>
 #include <wchar.h>
@@ -174,9 +176,11 @@
 #endif
 #endif
 
+#ifndef __vita__
 #ifdef  S_IFLNK
 #define SAVE_LINKS
 #endif
+#endif
 
 #if defined(__linux) || defined(__FreeBSD__)
 #include <sys/time.h>
@@ -225,7 +229,7 @@
 #endif
 
 // Unlike Apple x64, utimensat shall be available in all Apple M1 systems.
-#if _POSIX_C_SOURCE >= 200809L || defined(__APPLE__) && defined(__arm64__)
+#if (_POSIX_C_SOURCE >= 200809L || defined(__APPLE__) && defined(__arm64__)) && !defined(__vita__)
   #define UNIX_TIME_NS // Nanosecond time precision in Unix.
 #endif
 
diff --git a/ulinks.cpp b/ulinks.cpp
index 027991f..f5f75e6 100644
--- a/ulinks.cpp
+++ b/ulinks.cpp
@@ -22,6 +22,7 @@ static bool UnixSymlink(CommandData *Cmd,const std::string &Target,const wchar *
     }
     return false;
   }
+#ifndef __vita__
 #ifdef USE_LUTIMES
 #ifdef UNIX_TIME_NS
   timespec times[2];
@@ -38,6 +39,7 @@ static bool UnixSymlink(CommandData *Cmd,const std::string &Target,const wchar *
   tv[1].tv_usec=long(ftm->GetUnixNS()%1000000000/1000);
   lutimes(LinkNameA.c_str(),tv);
 #endif
+#endif
 #endif
 
   return true;
