diff --git a/src/compat/compat.c b/src/compat/compat.c
index 7890012..0149c41 100644
--- a/src/compat/compat.c
+++ b/src/compat/compat.c
@@ -520,7 +520,7 @@ void (*INT123_catchsignal(int signum, void(*handler)(int)))(int)
 
 #ifdef DONT_CATCH_SIGNALS
 	fprintf (stderr, "Not catching any signals.\n");
-	return ((void (*)()) -1);
+	return ((void (*)(int)) -1);
 #endif
 
 	new_sa.sa_handler = handler;
diff --git a/src/libmpg123/getcpuflags_arm.c b/src/libmpg123/getcpuflags_arm.c
index 8d0249a..920d600 100644
--- a/src/libmpg123/getcpuflags_arm.c
+++ b/src/libmpg123/getcpuflags_arm.c
@@ -19,6 +19,13 @@
 
 extern void INT123_check_neon(void);
 
+#ifdef __vita__
+unsigned int INT123_getcpuflags(struct cpuflags* cf)
+{
+	cf->has_neon = 1;
+	return 0;
+}
+#else
 #ifndef _M_ARM
 static sigjmp_buf jmpbuf;
 #else
@@ -65,3 +72,4 @@ unsigned int INT123_getcpuflags(struct cpuflags* cf)
 	
 	return 0;
 }
+#endif
