diff --git a/libxmp.pc.in b/libxmp.pc.in
index 665b6be..19de923 100644
--- a/libxmp.pc.in
+++ b/libxmp.pc.in
@@ -1,7 +1,7 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+prefix=${VITASDK}/arm-vita-eabi
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: libxmp
 Description: Xmp module player library
diff --git a/src/tempfile.c b/src/tempfile.c
index 6b05820..587d655 100644
--- a/src/tempfile.c
+++ b/src/tempfile.c
@@ -99,6 +99,14 @@ static int get_temp_dir(char *buf, size_t size)
 
 	return 0;
 }
+#elif defined __vita__
+
+static int get_temp_dir(char *buf, size_t size)
+{
+	strncpy(buf, "ux0:/data/tmp/", size);
+
+	return 0;
+}
 
 #else /* unix */
 
@@ -118,6 +126,31 @@ static int get_temp_dir(char *buf, size_t size)
 #endif
 
 
+#if defined __vita__
+
+FILE *make_temp_file(char **filename) {
+	char tmp[XMP_MAXPATH];
+	char tmpfilename[128];
+	FILE *tempfd;
+
+	if (get_temp_dir(tmp, XMP_MAXPATH) < 0)
+		return NULL;
+
+	snprintf(tmpfilename, 128, "xmp_%06x", rand() % 0xFFFFFF);
+	strncat(tmp, tmpfilename, XMP_MAXPATH - 10);
+
+	if ((*filename = libxmp_strdup(tmp)) == NULL)
+		return NULL;
+	if ((tempfd = fopen(tmp, "w+b")) == NULL)
+	{
+		free(*filename);
+	}
+
+	return tempfd;
+}
+
+# else
+
 FILE *make_temp_file(char **filename) {
 	char tmp[XMP_MAXPATH];
 	FILE *temp;
@@ -150,6 +183,8 @@ FILE *make_temp_file(char **filename) {
 	return NULL;
 }
 
+#endif
+
 /*
  * Windows doesn't allow you to unlink an open file, so we changed the
  * temp file cleanup system to remove temporary files after we close it
