diff --git a/SDLnetsys.h b/SDLnetsys.h
index 825d1cc..ba9247a 100644
--- a/SDLnetsys.h
+++ b/SDLnetsys.h
@@ -48,7 +48,9 @@
 #ifdef __FreeBSD__
 #include <sys/socket.h>
 #endif
+#ifndef __vita__
 #include <sys/ioctl.h>
+#endif
 #include <sys/time.h>
 #include <unistd.h>
 #include <fcntl.h>
@@ -58,7 +60,9 @@
 #endif
 #include <netinet/tcp.h>
 #include <sys/socket.h>
+#ifndef __vita__
 #include <net/if.h>
+#endif
 #include <netdb.h>
 #endif /* WIN32 */
 
