diff --git a/SDLnetsys.h b/SDLnetsys.h
index 6f6dfae..d515e6a 100644
--- a/SDLnetsys.h
+++ b/SDLnetsys.h
@@ -49,7 +49,9 @@ typedef int socklen_t;
 #ifdef __FreeBSD__
 #include <sys/socket.h>
 #endif
+#ifndef __vita__
 #include <sys/ioctl.h>
+#endif /* __vita__ */
 #include <sys/time.h>
 #include <unistd.h>
 #include <fcntl.h>
@@ -59,7 +61,9 @@ typedef int socklen_t;
 #endif
 #include <netinet/tcp.h>
 #include <sys/socket.h>
+#ifndef __vita__
 #include <net/if.h>
+#endif /* __vita__ */
 #include <netdb.h>
 #endif /* WIN32 */
 
