diff --git a/SDL2_mixerConfig.cmake.in b/SDL2_mixerConfig.cmake.in
index ad47ea1e..1fb4720f 100644
--- a/SDL2_mixerConfig.cmake.in
+++ b/SDL2_mixerConfig.cmake.in
@@ -54,10 +54,6 @@ if(EXISTS "${CMAKE_CURRENT_LIST_DIR}/SDL2_mixer-static-targets.cmake")
     include(CMakeFindDependencyMacro)
     include(PkgConfigHelper)
 
-    if(NOT DEFINED CMAKE_FIND_PACKAGE_PREFER_CONFIG)
-        set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ON)
-    endif()
-
     if(NOT SDL2MIXER_VENDORED)
         if(SDL2MIXER_FLAC_LIBFLAC AND NOT TARGET FLAC::FLAC)
             find_dependency(FLAC)
