diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54317572..2d5baaaa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -589,10 +589,10 @@ if(SDL2MIXER_MOD_XMP)
     if(SDL2MIXER_MOD_XMP_LITE)
         message(STATUS "Using system libxmp-lite")
         find_package(libxmp-lite REQUIRED)
-        set(tgt_xmp libxmp-lite::libxmp-lite)
+        set(tgt_xmp libxmp-lite)
         set(xmp_name libxmp-lite)
         if(NOT SDL2MIXER_MOD_XMP_SHARED)
-            list(APPEND PC_REQUIRES libxmplite)
+            list(APPEND PC_REQUIRES libxmp-lite)
         endif()
     else()
         message(STATUS "Using system libxmp")
diff --git a/cmake/Findlibxmp-lite.cmake b/cmake/Findlibxmp-lite.cmake
index d0b2bbbb..911ed77b 100644
--- a/cmake/Findlibxmp-lite.cmake
+++ b/cmake/Findlibxmp-lite.cmake
@@ -1,9 +1,10 @@
 find_library(libxmp_lite_LIBRARY
-    NAMES xmp
+    NAMES xmp-lite
 )
 
 find_path(libxmp_lite_INCLUDE_PATH
     NAMES xmp.h
+    PATH_SUFFIXES libxmp-lite
 )
 
 set(libxmp_lite_COMPILE_OPTIONS "" CACHE STRING "Extra compile options of libxmp_lite")
@@ -12,20 +13,19 @@ set(libxmp_lite_LINK_LIBRARIES "" CACHE STRING "Extra link libraries of libxmp_l
 
 set(libxmp_lite_LINK_FLAGS "" CACHE STRING "Extra link flags of libxmp_lite")
 
-find_package_handle_standard_args(libxmp_lite
+find_package_handle_standard_args(libxmp-lite
     REQUIRED_VARS libxmp_lite_LIBRARY libxmp_lite_INCLUDE_PATH
 )
 
 if(libxmp_lite_FOUND)
     if(NOT TARGET libxmp-lite::libxmp-lite)
         add_library(libxmp-lite::libxmp-lite UNKNOWN IMPORTED)
-        set_target_properties(libxmp_lite::libxmp_lite-shared PROPERTIES
+        set_target_properties(libxmp_lite::libxmp_lite PROPERTIES
             IMPORTED_LOCATION "${libxmp_lite_LIBRARY}"
             INTERFACE_INCLUDE_DIRECTORIES "${libxmp_lite_INCLUDE_PATH}"
             INTERFACE_COMPILE_OPTIONS "${libxmp_lite_COMPILE_OPTIONS}"
             INTERFACE_LINK_LIBRARIES "${libxmp_lite_LINK_LIBRARIES}"
             INTERFACE_LINK_FLAGS "${libxmp_lite_LINK_FLAGS}"
         )
-        endif()
     endif()
 endif()
