diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54317572..a066c586 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -796,8 +796,6 @@ if(SDL2MIXER_INSTALL)
         COMPONENT devel
     )
 
-    if(SDL2MIXER_BUILD_SHARED_LIBS)
-        # Only create a .pc file for a shared SDL2_mixer
         set(VERSION ${FULL_VERSION})
         set(SDL_VERSION ${SDL_REQUIRED_VERSION})
         set(prefix "${CMAKE_INSTALL_PREFIX}")
@@ -817,7 +815,7 @@ if(SDL2MIXER_INSTALL)
         else()
             set(PC_DESTDIR "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
         endif()
-        # Only install a SDL2_mixer.pc file in Release mode
+        # Always install SDL2_mixer.pc file: libraries might be different between config modes
         install(CODE "
             # FIXME: use file(COPY_FILE) if minimum CMake version >= 3.21
             execute_process(COMMAND \"\${CMAKE_COMMAND}\" -E copy_if_different
@@ -825,8 +823,7 @@ if(SDL2MIXER_INSTALL)
                 \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer.pc\")
             file(INSTALL DESTINATION \"\${CMAKE_INSTALL_PREFIX}/${PC_DESTDIR}\"
                 TYPE FILE
-                FILES \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer.pc\")" CONFIG Release COMPONENT devel)
-    endif()
+                FILES \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer.pc\")" COMPONENT devel)
 
     if(SDL2MIXER_BUILD_SHARED_LIBS AND (APPLE OR (UNIX AND NOT ANDROID)))
         install(
diff --git a/SDL2_mixer.pc.in b/SDL2_mixer.pc.in
index c14ffb65..a7ec8660 100644
--- a/SDL2_mixer.pc.in
+++ b/SDL2_mixer.pc.in
@@ -1,4 +1,4 @@
-prefix=@prefix@
+prefix=${VITASDK}/arm-vita-eabi
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
