diff --git a/vp8/common/generic/systemdependent.c b/vp8/common/generic/systemdependent.c
index 58c778aba..fbcb06336 100644
--- a/vp8/common/generic/systemdependent.c
+++ b/vp8/common/generic/systemdependent.c
@@ -38,7 +38,9 @@ static int get_cpu_count(void) {
   int core_count = 16;
 
 #if HAVE_UNISTD_H
-#if defined(_SC_NPROCESSORS_ONLN)
+#if defined(__vita__)
+  core_count = 3;
+#elif defined(_SC_NPROCESSORS_ONLN)
   core_count = (int)sysconf(_SC_NPROCESSORS_ONLN);
 #elif defined(_SC_NPROC_ONLN)
   core_count = (int)sysconf(_SC_NPROC_ONLN);
