diff -Naur libarchive-3.3.2.orig/libarchive/archive_ppmd7.c libarchive-3.3.2/libarchive/archive_ppmd7.c
--- libarchive-3.3.2.orig/libarchive/archive_ppmd7.c	2017-07-10 04:38:04.000000000 +0200
+++ libarchive-3.3.2/libarchive/archive_ppmd7.c	2022-05-12 13:03:24.092075180 +0200
@@ -4,7 +4,9 @@
 
 #include "archive_platform.h"
 
+#ifndef __vita__
 #include <memory.h>
+#endif
 
 #include "archive_ppmd7_private.h"
 
diff -Naur libarchive-3.3.2.orig/libarchive/archive_read_disk_posix.c libarchive-3.3.2/libarchive/archive_read_disk_posix.c
--- libarchive-3.3.2.orig/libarchive/archive_read_disk_posix.c	2017-07-10 04:38:04.000000000 +0200
+++ libarchive-3.3.2/libarchive/archive_read_disk_posix.c	2022-05-12 13:03:54.928289021 +0200
@@ -100,8 +100,10 @@
 #include "archive_read_disk_private.h"
 
 #ifndef HAVE_FCHDIR
+#ifndef __vita__
 #error fchdir function required.
 #endif
+#endif
 #ifndef O_BINARY
 #define O_BINARY	0
 #endif
diff -Naur libarchive-3.3.2.orig/libarchive/archive_read_support_filter_program.c libarchive-3.3.2/libarchive/archive_read_support_filter_program.c
--- libarchive-3.3.2.orig/libarchive/archive_read_support_filter_program.c	2017-07-10 04:38:04.000000000 +0200
+++ libarchive-3.3.2/libarchive/archive_read_support_filter_program.c	2022-05-12 13:04:33.956559513 +0200
@@ -58,7 +58,13 @@
 #include "archive_read_private.h"
 #include "filter_fork.h"
 
-
+#ifdef __vita__
+int
+__archive_read_program(struct archive_read_filter *self, const char *cmd)
+{
+	return (ARCHIVE_OK);
+}
+#else
 #if ARCHIVE_VERSION_NUMBER < 4000000
 /* Deprecated; remove in libarchive 4.0 */
 int
@@ -516,3 +522,4 @@
 
 	return (e);
 }
+#endif
diff -Naur libarchive-3.3.2.orig/libarchive/archive_read_support_format_mtree.c libarchive-3.3.2/libarchive/archive_read_support_format_mtree.c
--- libarchive-3.3.2.orig/libarchive/archive_read_support_format_mtree.c	2017-07-10 04:38:04.000000000 +0200
+++ libarchive-3.3.2/libarchive/archive_read_support_format_mtree.c	2022-05-12 13:05:34.912981657 +0200
@@ -1207,7 +1207,11 @@
 				mtree->fd = -1;
 				st = NULL;
 			}
+#ifdef __vita__
+		} else if (stat(path, st) == -1) {
+#else
 		} else if (lstat(path, st) == -1) {
+#endif
 			st = NULL;
 		}
 
diff -Naur libarchive-3.3.2.orig/libarchive/archive_write_disk_posix.c libarchive-3.3.2/libarchive/archive_write_disk_posix.c
--- libarchive-3.3.2.orig/libarchive/archive_write_disk_posix.c	2017-07-10 04:38:04.000000000 +0200
+++ libarchive-3.3.2/libarchive/archive_write_disk_posix.c	2022-05-12 13:06:04.529186629 +0200
@@ -2527,8 +2527,10 @@
 #if !defined(HAVE_LSTAT)
 	/* Platform doesn't have lstat, so we can't look for symlinks. */
 	(void)path; /* UNUSED */
+#ifndef __vita__
 	(void)error_number; /* UNUSED */
 	(void)error_string; /* UNUSED */
+#endif
 	(void)flags; /* UNUSED */
 	return (ARCHIVE_OK);
 #else
diff -Naur libarchive-3.3.2.orig/libarchive/archive_write_set_format_iso9660.c libarchive-3.3.2/libarchive/archive_write_set_format_iso9660.c
--- libarchive-3.3.2.orig/libarchive/archive_write_set_format_iso9660.c	2017-07-10 04:38:04.000000000 +0200
+++ libarchive-3.3.2/libarchive/archive_write_set_format_iso9660.c	2022-05-12 13:06:55.245537452 +0200
@@ -2178,8 +2178,10 @@
 	strncpy(system_id, "Windows", size-1);
 	system_id[size-1] = '\0';
 #else
+#ifndef __vita__
 #error no way to get the system identifier on your platform.
 #endif
+#endif
 }
 
 static void
@@ -2505,7 +2507,7 @@
 
 #if defined(HAVE__GET_TIMEZONE)
 	_get_timezone(&offset);
-#elif defined(__CYGWIN__) || defined(__MINGW32__) || defined(__BORLANDC__)
+#elif defined(__CYGWIN__) || defined(__MINGW32__) || defined(__BORLANDC__) || defined(__vita__)
 	offset = _timezone;
 #else
 	offset = timezone;
diff -Naur libarchive-3.3.2.orig/libarchive_fe/passphrase.c libarchive-3.3.2/libarchive_fe/passphrase.c
--- libarchive-3.3.2.orig/libarchive_fe/passphrase.c	2017-07-10 04:38:04.000000000 +0200
+++ libarchive-3.3.2/libarchive_fe/passphrase.c	2022-05-12 13:08:41.562272222 +0200
@@ -129,9 +129,12 @@
 #endif
 #include <signal.h>
 #include <string.h>
+#ifndef __vita__
 #include <termios.h>
+#endif
 #include <unistd.h>
 
+#ifndef __vita__
 #ifndef _PATH_TTY
 #define _PATH_TTY "/dev/tty"
 #endif
@@ -303,6 +306,13 @@
 		errno = save_errno;
 	return(nr == -1 ? NULL : buf);
 }
+#else
+static char *
+readpassphrase(const char *prompt, char *buf, size_t bufsiz, int flags)
+{
+  return NULL;
+}
+#endif
 #endif /* _WIN32 && !__CYGWIN__ */
 #endif /* HAVE_READPASSPHRASE */
 
